reader.go

```diff
@@ -30,48 +31,77 @@ func (o *ReaderOptions) norm(name string) {
 
 // Reader reads data from a remote feed.
 type Reader struct {
-	remote *bfs.Object
-	opt    ReaderOptions
-	ctx    context.Context
-	num    int
+	ctx context.Context
+	opt ReaderOptions
 
-	br io.ReadCloser // bfs reader
-	cr io.ReadCloser // compression reader
-	fd FormatDecoder
+	remotes []*bfs.Object
+	cur     *streamReader
+	fd      FormatDecoder
```

Comment:
I now wonder if it would have been better/less messy to keep this on `streamReader`